benchmarks/src/main/java/run/chronicle/staged/StagedPerformanceMain.java

```diff
@@ -0,0 +1,269 @@
+                    appender0.pretouch();
+                    Jvm.pause(10);
+                }
+            });
+            pretoucher.setDaemon(true);
+            pretoucher.start();
+
+            for (int i = 1; i <= count; i++) {
+                long end = start + (long) intervalNS * i;
+                while (System.nanoTime() <= end)
+                    Thread.yield();
+
+                try (DocumentContext dc = appender.writingDocument()) {
+                    Wire wire = dc.wire();
+                    wire.write("data");
+                    Bytes<?> bytes = wire.bytes();
+                    long wp = bytes.writePosition();
+                    int paddingToAdd = (int) (-wp & 7);
+                    wire.addPadding(paddingToAdd);
+                    datum.bytesStore(bytes, bytes.writePosition(), datumSize);
+                    bytes.writeSkip(datumSize);
+
+                    // write the data
+                    populateDatum(count, datum, i);
+                }
+            }
+            pretoucher.interrupt();
+            try {
+                pretoucher.join(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+        long time = System.nanoTime() - start;
+        if (!WARMUP) {
+            System.out.println("Producer wrote " + count + " messages in " + time / 1000 / 1e6 + " seconds");
+        }
+    }
+
+    private static void populateDatum(int count, IFacadeAll datum, int i) {
+        final int left = count - i;
+        if (left % REPORT_INTERVAL == 0 & !WARMUP & left > 0)
+            System.out.println("producer " + left + " left");
+        datum.setValue3(left); // the number remaining.
+        datum.setTimestampAt(0, System.nanoTime());
+    }
+
+    public static void firstStage() {
+        IFacadeAll datum = Values.newNativeReference(IFacadeAll.class);
+        long datumSize = datum.maxSize();
+//        int count = 0;
+//        long start = System.nanoTime();
+        try (ExcerptTailer tailer = q0.createTailer();
+             LongValue value = q0.indexForId("stage1")) {
+
+            while (true) {
+                final long index;
+                try (final DocumentContext dc = tailer.readingDocument()) {
+                    if (!dc.isPresent()) {
+                        Jvm.nanoPause();
+                        continue;
+                    }
+
+                    applyFlyweight(datum, datumSize, dc);
+
+                    processStage(datum, 1);
+                    index = dc.index();
+                }
+                // write the index processed
+                value.setOrderedValue(index);
+//                count++;
+                final int left = datum.getValue3();
+//                if (left % REPORT_INTERVAL == 0 & !WARMUP & left > 0)
+//                    System.out.println("stage " + 1 + ", left " + left);
+                if (left == 0) {
+//                    long time = System.nanoTime() - start;
```

Comment:
please either remove the commented-out code, or else enable with a system property